backend/models/helper.js

```diff
@@ -0,0 +1,78 @@
+const bcrypt = require('bcrypt');
+const Homeowner = require('../models/homeownerModel');
+const Contractor = require('../models/contractorModel');
+const validator = require('validator');
+
+async function login(user) {
+    const { email, password } = user;
+    if (!email || !password) {
+        throw Error('All fields must be filled');
+    }
+
+    const userInfo = await getUserData({ email });
+
+    if (!userInfo) {
+        throw Error('Account does not exist. Please signup first.');
+    } 
+
+    return userInfo;
+}
+
+
+async function signup(user) {
+    const { email, password, name, userType } = user;
+    //validate
+    if (!name || !email || !password) {
+        throw Error('All fields must be filled');
+    }
+
+    if (!validator.isEmail(email)){
+        throw Error('Email is not valid')
+    }
+
+    if (!validator.isStrongPassword(password)) {
+        throw Error('Password is not strong enough')
+    }
+
+    const existsInHomeowner = await exists({ email }, Homeowner);
+    const existsInContractor = await exists({ email }, Contractor);
+
+    if (existsInHomeowner || existsInContractor){
+        throw Error("Email is already in use");
+    }
+    
+    return (userType === "Homeowner")
+        ? storeUser(email, password, name, Homeowner)
+        : storeUser(email, password, name, Contractor);
+
+}
+
+async function exists(queryObj, model) {
+    const data = await model.findOne(queryObj).lean();
+    return data;   
+}
+
+async function storeUser(email, password, name, model) {
+    //hash password
+    const salt = await bcrypt.genSalt(10);
```

Comment:
Any particular reason for the choice of 10? This looks good, just curious.

Reply:
10 is the default value for genSalt(), meaning there will be 10 rounds for salting. Too much salting will overcomplicate the hash algorithm and too little will be insecure. For our server, using the default might be the best choice

Reply:
I think it shouldn't be too big that it takes too much time to hash/unhash nor too small that it's easy to get hacked - I believe 10 is the default option and is somewhere in the middle